test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -161,6 +163,20 @@ private Map<String, Object> getPravegaOnlyDeployment(String zkLocation, int cont
                 .build();
     }
 
+    private Object getPodAntiAffinity() {
```

Comment:
My understanding of this approach is that, during system tests, the restarts of Segment Store pods make sometimes Zookeeper pods to get evicted to get enough resources and this causes unexpected restarts that are marked as test failures. With this policy, the idea is to prevent as much as possible this situation to happen, right? Have you monitored the location of Segment Store and Zookeeper pods to check up to which point this policy is enforced?